test/E2ETest/Infrastructure/EqualityYamlNodeVisitor.cs

```diff
@@ -0,0 +1,136 @@
+            }
+
+            if (node.Tag?.Equals(otherNode.Tag) == false)
+            {
+                throw new TyeYamlException($"Expected tags to be equal, difference starting at Expected: ({node.Start.Line}, {node.Start.Column}) Actual: ({otherNode.Start.Line}, {otherNode.Start.Column})");
+            }
+
+            if (node.Children.Count != otherNode.Children.Count)
+            {
+                throw new TyeYamlException($"Number of children differ for sequence, Expected: ({node.Start.Line}, {node.Start.Column}) Actual: ({otherNode.Start.Line}, {otherNode.Start.Column})");
+            }
+
+            for (var i = 0; i < node.Children.Count; ++i)
+            {
+                var childNode = node.Children[i];
+                var otherChildNode = otherNode.Children[i];
+
+                if (childNode.NodeType != otherChildNode.NodeType)
+                {
+                    throw new TyeYamlException($"Child node types differ, Expected: ({childNode.Start.Line}, {childNode.Start.Column}) Actual: ({otherChildNode.Start.Line}, {otherChildNode.Start.Column})");
+                }
+
+                if (childNode.NodeType == YamlNodeType.Mapping)
+                {
+                    VisitMapping((YamlMappingNode)childNode, (YamlMappingNode)otherChildNode);
+                }
+                else if (childNode.NodeType == YamlNodeType.Scalar)
+                {
+                    VisitScalar((YamlScalarNode)childNode, (YamlScalarNode)otherChildNode);
+                }
+                else if (childNode.NodeType == YamlNodeType.Sequence)
+                {
+                    VisitSequence((YamlSequenceNode)childNode, (YamlSequenceNode)otherChildNode);
+                }
+            }
+        }
+
+        public void VisitMapping(YamlMappingNode node, YamlMappingNode otherNode)
+        {
+            if (otherNode == null)
+            {
+                throw new TyeYamlException($"Expected mapping to be not null, difference starting at Expected: ({node.Start.Line}, {node.Start.Column})");
+            }
+
+            if (node.Tag?.Equals(otherNode.Tag) == false)
+            {
+                throw new TyeYamlException($"Expected tags to be equal, difference starting at Expected: ({node.Start.Line}, {node.Start.Column}) Actual: ({otherNode.Start.Line}, {otherNode.Start.Column})");
+            }
+
+            if (node.Children.Count != otherNode.Children.Count)
+            {
+                throw new TyeYamlException($"Number of children differ for mapping, Expected: ({node.Start.Line}, {node.Start.Column}) Actual: ({otherNode.Start.Line}, {otherNode.Start.Column})");
+            }
+
+            foreach (var child in node.Children)
+            {
+                var childValue = child.Value;
+                var otherChildValue = otherNode.Children[child.Key];
```

Comment:
This will also blow up with a bad error in the event that the expected mapping isn't present in `otherNode`. Can we do a `TryGetValue` instead?